enhancements/okd/okd-featureset.md

```diff
@@ -0,0 +1,347 @@
+---
+title: okd-featureset
+authors:
+  - "@jatinsu"
+reviewers:
+  - "@JoelSpeed, for API review and feature gate validation"
+  - "@Prashanth684, for OKD-specific implementation details"
+  - "@sdodson, for OKD platform considerations"
+  - "@wking, for the Cluster Version Operator"
+approvers:
+  - "@sdodson"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2025-12-02
+last-updated: 2025-12-02
+tracking-link:
+  - https://issues.redhat.com/browse/OKD-259
+see-also:
+  - "/enhancements/installer/feature-sets.md"
+replaces:
+superseded-by:
+---
+
+# OKD Feature Set
+
+## Summary
+
+This enhancement introduces a new "OKD" feature set in the OpenShift API that will be enabled by default on all OKD clusters. This feature set allows OKD clusters to enable select TechPreview features in addition to all features in the Default feature set while maintaining upgrade capability. This feature set will replace the Default feature set for OKD clusters.
+
```

Comment:
in the summary and in the motivation we should mention that one of the main drivers is to enable features which have stable APIs early on OKD to not only provide these features to the community but also as a signal driver for OCP (with the community trying out upgrades, adopting features and our own nightly jobs testing upgrades and e2es)